### PR TITLE
Feat: definition support controller requirement

### PR DESCRIPTION
**Pull request by yangsoon**
Signed-off-by: yangsoon <songyang.song@alibaba-inc.com>

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by Somefive — APPROVED** [review]

File: pkg/oam/labels.go [hunk 1]
```diff
@@ -211,6 +211,9 @@ const (
 	// AnnotationControllerRequirement indicates the controller version that can process the application.
 	AnnotationControllerRequirement = "app.oam.dev/controller-version-require"
 
+	// AnnotationDefinitionControllerRequirement indicates the controller version that can process the definition.
+	AnnotationDefinitionControllerRequirement = "definition.oam.dev/controller-version-require"
```
Comment (wonderflow) [inline pkg/oam/labels.go:215]: let's use the same annotation `AnnotationControllerRequirement`, can we?

Comment (yangsoon) [reply pkg/oam/labels.go:215]: `AnnotationControllerRequirement` is used for application CR

Comment (wonderflow) [reply pkg/oam/labels.go:215]: can they be the same one?

Comment (yangsoon) [reply pkg/oam/labels.go:215]: v.1.2.x already uses this feature, if it is changed to the same name, there will be incompatibility problems

**Review by wonderflow — APPROVED** [review]

**Pull request merged**